### PR TITLE
Purge command and new networking features

### LINKED ISSUE #7: [Feature] purge command
Hi,

I'm posting this here, as my request on discord doesn't really help keeping track of stuff haha.

 It would be awesome to have access to a `ledger purge [args]` command, that would accept arguments that the search command provide. So we could remove everything older than 30days for example, or everything done by a banned user.

### PR DESCRIPTION
**Pull request by yitzy299**
Closes #7 and #63

### CODE REVIEW

File: docs/commands/purge.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+`/ledger purge`
+Permission: `ledger.commands.purge`
+
+---
+
+### `/ledger purge <params>`
+This command will remove all entries from the database which fit the parameters specified. See [parameters](parameters.md)
+
+
+**Inspect Mode**
```
Comment (PotatoPresident) [inline docs/commands/purge.md:11]: 🤔

File: src/main/kotlin/com/github/quiltservertools/ledger/commands/subcommands/PurgeCommand.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            .requires(Permissions.require("ledger.commands.purge", config[SearchSpec.purgePermissionLevel]))
+            .then(SearchParamArgument.argument(CommandConsts.PARAMS).executes {
+                runPurge(it, SearchParamArgument.get(it, CommandConsts.PARAMS))
+            })
+            .build()
+    }
+
+    private fun runPurge(ctx: CommandContext<ServerCommandSource>, params: ActionSearchParams): Int {
+        val source = ctx.source
+        source.sendFeedback(TranslatableText("text.ledger.purge.starting").setStyle(TextColorPallet.secondary), false)
```
Comment (PotatoPresident) [inline src/main/kotlin/com/github/quiltservertools/ledger/commands/subcommands/PurgeCommand.kt:32]:
```suggestion
        source.sendFeedback(TranslatableText("text.ledger.purge.starting").setStyle(TextColorPallet.secondary), true)
```

File: src/main/kotlin/com/github/quiltservertools/ledger/database/DatabaseManager.kt [hunk 3]
```diff
@@ -417,4 +425,65 @@ object DatabaseManager {
...
+            }
+
+            if (params.min != null && params.max != null) {
+                where.and(Tables.Actions.x.between(params.min.x, params.max.x))
+                where.and(Tables.Actions.y.between(params.min.y, params.max.y))
+                where.and(Tables.Actions.z.between(params.min.z, params.max.z))
+            }
+
+            if (params.time != null) {
+                // Leaving here to annoy potato
```
Comment (PotatoPresident) [inline src/main/kotlin/com/github/quiltservertools/ledger/database/DatabaseManager.kt:456]: 😡

File: src/main/kotlin/com/github/quiltservertools/ledger/config/SearchSpec.kt [hunk 4]
```diff
@@ -4,4 +4,5 @@ import com.uchuhimo.konf.ConfigSpec
 
 object SearchSpec : ConfigSpec() {
     val pageSize by required<Int>()
+    val purgePermissionLevel by required<Int>()
```
Comment (PotatoPresident) [inline src/main/kotlin/com/github/quiltservertools/ledger/config/SearchSpec.kt:7]: This would fit better in DatabaseSpec or somewhere else I think

**Review by PotatoPresident — CHANGES_REQUESTED** [review]
Didn't look at the actual database stuff yet as I'm on mobile

File: src/main/kotlin/com/github/quiltservertools/ledger/config/SearchSpec.kt (hunk 4 above)
Comment (yitzy299) [reply src/main/kotlin/com/github/quiltservertools/ledger/config/SearchSpec.kt:7]: Probably, yes

**Review by PotatoPresident — APPROVED** [review]

**Pull request merged**